nixos/modules/services/misc/devpi-server.nix

```
@@ -0,0 +1,125 @@
+{
+  pkgs,
+  lib,
+  config,
+  ...
+}:
+with lib;
+let
+  cfg = config.services.devpi-server;
+  serverDir = "/var/lib/devpi";
+in
+{
+  options.services.devpi-server = {
+    enable = mkEnableOption (mdDoc "Devpi Server");
+
+    package = mkOption {
+      type = with types; package;
+      default = pkgs.devpi-server;
+      defaultText = lib.literalExpression "pkgs.devpi-server";
+      description = "Which devpi-server package to use.";
+    };
+
+    primaryUrl = mkOption {
+      type = types.str;
+      description = "Url for the primary node. Required option for replica nodes.";
+    };
+
+    replica = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Run node as a replica.
+        Requires the secretFile option and the primaryUrl to be enabled.
+      '';
+    };
+
+    secretFile = mkOption {
+      type = types.str;
+      default = "";
+      description = ''
+        Path to a shared secret file used for synchronization,
+        Required for all nodes in a replica/primary setup.
+      '';
+    };
+
+    host = mkOption {
+      type = types.str;
+      default = "localhost";
+      description = ''
+        domain/ip address to listen on
+      '';
+    };
+
+    port = mkOption {
+      type = types.port;
+      default = 3141;
+      description = "The port on which Devpi Server will listen.";
+    };
+
+    openFirewall = mkOption {
+      type = types.bool;
+      default = false;
+      description = "Opens the port used by the firewall.";
+    };
+  };
+
+  config = mkIf cfg.enable {
+
+    systemd.services.devpi-server = {
```

```suggestion
    systemd.services.devpi-server = {
      description = "devpi PyPI-compatible server";
      documentation = [ "https://devpi.net/docs/devpi/devpi/stable/+d/index.html" ];
```

We can also improve the SystemD service with some description, documentation.
And also some SystemD hardenings like the Debian team has done.

See https://salsa.debian.org/python-team/packages/devpi-server/-/blob/debian/master/debian/devpi-server.service

What do you think?

---

Good idea, I'll take most of these, except `ReadWritePaths` since I think that might cause issues.